### ISSUE TITLE
Make element go full width if within limited with parent

### ISSUE THREAD

**Issue by m0n0mind** [uikit/uikit#3155]
Why not add a possibility to make containers within parents with limited/defined width go full width (of the viewport) in order to easily add sections that break out of the parents' width context, as proposed on css-tricks:

https://css-tricks.com/full-width-containers-limited-width-parents/

**Comment by saschadube** [comment]
As far as I know 100vw does not take the scrollbar into account, right?

**Comment by saschadube** [comment]
@TomS- Thanks for the example. [Here](https://codepen.io/anon/pen/RxEBQV) is the vertical scrollbar I meant.

**Comment by TomS-** [comment]
@saschadube I guess you could do https://codepen.io/anon/pen/MrZzzd 

If this was to make it into the core though;
I feel like JavaScript is the most dynamic way to achieve this in case someone wants to change the width of the container without the need to recompile, however it could be the case that a vertical scrollbar will appear slightly before the JavaScript loads. Since UIKit JS is loaded into the head, and since it does all the magic with uk-grid etc. I feel like JavaScript might be the way to go.

**Comment by m0n0mind** [comment]
thanks @TomS- for the example.
But if it's possible regarding modern browser support a pure css solution is preferred.

@saschadube the browser bugs are not yet fixed.

But it seems we can workaround this by using 
`html { overflow-x: hidden }`

see https://codepen.io/anon/pen/RxEEbx

read more about it here:
https://cloudfour.com/thinks/breaking-out-with-viewport-units-and-calc/#one-big-dumb-caveat
or here:
https://codepen.io/tigt/post/bust-elements-out-of-containers-with-one-line-of-css#oh-no-a-horizontal-scrollbar-6
or in the w3c specs:
https://www.w3.org/TR/css3-values/#viewport-relative-lengths

**Comment by TomS-** [comment]
@m0n0mind I agree CSS is preferred, it can use the less variables to work it out. It should also check whether it's in a container or not. However, you can't always guarantee that it will be in a container. I still stand by JavaScript is the most dynamic way and means you can put full with elements anywhere if so required.

**Comment by m0n0mind** [comment]
@TomS- Yeah, suppose you're right.
Either way, as a JS-enhanced solution or as pure CSS, it would be a good addition to uikit core.
